lib/chef/knife/ec2_server_create.rb

```diff
@@ -1054,6 +1065,9 @@ def create_server_def
           server_def[:block_device_mapping] = (server_def[:block_device_mapping] || []) << {'VirtualName' => "ephemeral#{i}", 'DeviceName' => device_name}
         end
 
+        ## Termination Protection cannot be enabled for spot instances ##
+        server_def[:disable_api_termination] = locate_config_value(:disable_api_termination) if locate_config_value(:spot_price).nil?
```

Comment:
@btm this check is just to prevent `disable-api-termination` option (irrespective of its value) from being getting passed in the `server_definition` of `spot instance request` as `spot instance` does not support this feature. The `disable-api-termination` option will only get passed in the `server_definition` of `non-spot instance request`. Also, even if user set this option `disable-api-termination` in a `spot instance request` then he/she will get error as `spot-price and disable-api-termination options cannot be passed together as 'Termination Protection' cannot be enabled for spot instances.`. You can check this validation code [here](https://github.com/chef/knife-ec2/pull/427/files#diff-e2993b8367db674f58a052036f00052aR900).